payu/experiment.py

```diff
@@ -596,6 +638,17 @@ def run(self, *user_flags):
 
                 shutil.copyfile(src, dest)
 
+            for fname in (self.job_fname, self.env_fname):
+
+                src = os.path.join(self.control_path, fname)
+
+                # Assumes .yaml extension
+                dest = os.path.join(error_log_dir,
+                                    fname[:-5] + '.' + job_id + fname[-5:])
```

Comment:
Consider using `splitext` here, and maybe bail if it's not a YAML file (unless there's some case where it ought to work?)

Something like this:
```
base, ext = os.path.splitext(fname)
assert(ext == '.yaml')
job_fname = ''.join(base, '.', job_id, ext)
dest = os.path.join(error_log_dir, job_fname)
```

Reply:
Hah! Just did this before I noticed your comment.